### PR TITLE
Feature/name id assignment

### PR DESCRIPTION
**Pull request by stevemanion**
_To describe..._

### CODE REVIEW

File: data/meta/record-linking/snapshot-change-logs/05-14-to-05-15-change-log.csv [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+FORMER,Mellissa,Confirmed,08/04/2020,Female,20 to 29,Waitematā,Yes,Canada,NZ23,null,23/03/2020
+UPDATE,Mellissa,Confirmed,08/04/2020,Female,20 to 29,Waitematā,Yes,Canada,NZ23,20/03/2020,21/03/2020
+FORMER,Wilma,Probable,11/04/2020,Female,20 to 29,Auckland,Yes,United Kingdom,QR920,26/03/2020,27/03/2020
+UPDATE,Wilma,Probable,11/04/2020,Female,20 to 29,Auckland,Yes,Qatar,QR920,26/03/2020,27/03/2020
+FORMER,Alexia,Confirmed,26/03/2020,Female,60 to 69,Counties Manukau,Yes,Qatar,null,null,17/03/2020
+UPDATE,Alexia,Confirmed,26/03/2020,Female,60 to 69,Counties Manukau,Yes,Qatar,QR920,15/03/2020,17/03/2020
+FORMER,Roland,Confirmed,01/04/2020,Male,20 to 29,Waikato,Yes,New Zealand,EK488,23/03/2020,24/03/2020
+UPDATE,Roland,Confirmed,01/04/2020,Male,20 to 29,Waikato,Yes,United Arab Emirates,EK488,23/03/2020,24/03/2020
+FORMER,Sylvia,Confirmed,22/03/2020,Female,60 to 69,Canterbury,Yes,Egypt,NZ884,18/03/2020,18/03/2020
+UPDATE,Sylvia,Confirmed,26/03/2020,Female,60 to 69,Canterbury,Yes,Egypt,NZ884,18/03/2020,18/03/2020
```
Comment (stevemanion) [inline data/meta/record-linking/snapshot-change-logs/05-14-to-05-15-change-log.csv:12]: @whistlebee conflation of interest 🤓

**Comment by whistlebee** [comment]
What on earth is going on here haha
<details>
<summary>Show</summary>
<p>

| nameId   | log    | reportDate   | dhb                | overseas   | lastCountry                               | flightNumber   | departureDate   | arrivalDate   | f                             |
|:---------|:-------|:-------------|:-------------------|:-----------|:------------------------------------------|:---------------|:----------------|:--------------|:------------------------------|
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | nan        | nan                                       | nan            | 5/03/2020       | nan           | 03-29-to-03-30-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | No         | nan                                       | nan            | 5/03/2020       | nan           | 03-29-to-03-30-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | No         | nan                                       | nan            | 5/03/2020       | nan           | 03-31-to-04-01-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 03-31-to-04-01-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-01-to-04-02-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-01-to-04-02-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-03-to-04-04-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-03-to-04-04-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-04-to-04-05-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-04-to-04-05-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-06-to-04-08-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | nan            | 5/03/2020       | 5/03/2020     | 04-06-to-04-08-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | nan            | 5/03/2020       | 5/03/2020     | 04-13-to-04-14-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-13-to-04-14-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-14-to-04-15-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-14-to-04-15-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | Cathay Pacific | 5/03/2020       | 5/03/2020     | 04-15-to-04-16-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | nan            | 5/03/2020       | 5/03/2020     | 04-15-to-04-16-change-log.csv |
| Everett  | FORMER | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong (Special Administrative Region) | nan            | 5/03/2020       | 5/03/2020     | 04-16-to-04-17-change-log.csv |
| Everett  | UPDATE | 25/03/2020   | Nelson Marlborough | Yes        | Hong Kong                                 | nan            | 5/03/2020       | 5/03/2020     | 04-16-to-04-17-change-log.csv |

</p>
</details>

**Comment by stevemanion** [comment]
> What on earth is going on here haha
> 
> Show
> nameId	log	reportDate	dhb	overseas	lastCountry	flightNumber	departureDate	arrivalDate	f
> Everett	FORMER	25/03/2020	Nelson Marlborough	nan	nan	nan	5/03/2020	nan	03-29-to-03-30-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	No	nan	nan	5/03/2020	nan	03-29-to-03-30-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	No	nan	nan	5/03/2020	nan	03-31-to-04-01-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	03-31-to-04-01-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-01-to-04-02-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	Cathay Pacific	5/03/2020	5/03/2020	04-01-to-04-02-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	Cathay Pacific	5/03/2020	5/03/2020	04-03-to-04-04-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	Cathay Pacific	5/03/2020	5/03/2020	04-03-to-04-04-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong	Cathay Pacific	5/03/2020	5/03/2020	04-04-to-04-05-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-04-to-04-05-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-06-to-04-08-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	nan	5/03/2020	5/03/2020	04-06-to-04-08-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	nan	5/03/2020	5/03/2020	04-13-to-04-14-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-13-to-04-14-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-14-to-04-15-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	Cathay Pacific	5/03/2020	5/03/2020	04-14-to-04-15-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong	Cathay Pacific	5/03/2020	5/03/2020	04-15-to-04-16-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	nan	5/03/2020	5/03/2020	04-15-to-04-16-change-log.csv
> Everett	FORMER	25/03/2020	Nelson Marlborough	Yes	Hong Kong (Special Administrative Region)	nan	5/03/2020	5/03/2020	04-16-to-04-17-change-log.csv
> Everett	UPDATE	25/03/2020	Nelson Marlborough	Yes	Hong Kong	nan	5/03/2020	5/03/2020	04-16-to-04-17-change-log.csv

@whistlebee looks like you've ingested it all into one big change history, nice! It's pretty wild huh? I remember Everett changing a lot between snapshots. The case details are very unique, so I don't think it is a potential conflation. However over the snapshots it's as if information about this case is coming from multiple sources (`flightNumber` and `lastCountry` toggle a lot). I'm sure you can find many more cases like this, as I remember some name ids cropping up again and again.

Now everything is linked it will be interesting to see what we can discover next! I would like to do some higher level data integrity checks too in order to catch records such as the Greg Mortimer ones that keep getting added and removed as well!

**Comment by stevemanion** [comment]
Merging to finally bring this work in!

**Pull request merged**